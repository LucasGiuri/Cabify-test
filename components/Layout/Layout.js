import Header from '../../components/Header/Header';
import './Layout.scss';

const Layout = ({children}) => {
	return (
		<div className="layout">
			<div className="layout__navigation">
				<Header />
			</div>
			<div className="layout__content">
				{children}
			</div>
		</div>
	);
}

export default Layout;