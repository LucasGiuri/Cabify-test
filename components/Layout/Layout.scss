@import '../../scss/breakpoints';
@import '../../scss/colors';

.layout {
	display: flex;
	flex-direction: column;
	justify-content: center;
	position:relative;
	width: 100%;
	font-family: 'Roboto', sans-serif;
}

.layout__navigation {
	width: 100%;
	background: $white;
	position: relative;
}

.layout__content {
	max-width: 100%;
	padding-left: 20px;
	padding-right: 20px;
	margin: 0 auto;
}

@media #{breakpoint('medium')} {
	.layout__content {
		width: 50em;
	}
}