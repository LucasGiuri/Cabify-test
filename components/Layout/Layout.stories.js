import React from 'react';
import { storiesOf } from '@storybook/react';
import Layout from './Layout';

storiesOf('Layout')
  .add('Layout default', () => (
    <Layout>
        <h1>Fake Layout</h1>
    </Layout>
  ))
