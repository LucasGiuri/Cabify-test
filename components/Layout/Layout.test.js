import React from 'react';
import renderer from 'react-test-renderer';
import Layout from './Layout';

jest.mock('../../components/Header/Header', () => 'Header');

describe('Layout', () => {
  it('Layout basic render snapshoot', () => {
    const component = renderer.create(<Layout> <h1>Fake content </h1></Layout>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
