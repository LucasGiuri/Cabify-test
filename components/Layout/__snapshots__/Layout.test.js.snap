// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Layout basic render snapshoot 1`] = `
<div
  className="layout"
>
  <div
    className="layout__navigation"
  >
    <Header />
  </div>
  <div
    className="layout__content"
  >
     
    <h1>
      Fake content 
    </h1>
  </div>
</div>
`;
